lacci/lib/shoes/app.rb

```diff
@@ -120,7 +122,10 @@ def init
       send_shoes_event(event_name: "init")
       return if @do_shutdown
 
-      with_slot(@document_root, &@app_code_body)
+      with_slot(@document_root) do
+        @content_container = flow(width: 1.0, height: 1.0)
+        with_slot(@content_container, &@app_code_body)
```

Comment:
This automatically makes a new flow around every app?

Reply:
Ah, okay. There's a new flow that then contains whatever the page is. Hrm. I should figure out if that messes with the formatting. I know with CSS stuff we had a devil of a time figuring out when adding a new container to the tree changed formatting. But this one may be fine.

Reply:
Yes I was having bugs where certain content was getting painted over and over on button click. Part of resolving this, and I thought this made sense, was a silent container for each and every "page"

Reply:
Hm. Since you're clearing the content_container every time, I'd think it'd make more sense to clear the DocumentRoot. If you're getting paint errors with that method, we should figure out why and fix those. Also, can you skip the stack() around pages if you're doing this? It seems odd to me that pages stack vertically by default but everything else stacks horizontally. And I *think* you can just skip making the extra container.